frontend-cert/js-projects/dates-lab/user-stories.md

```diff
@@ -0,0 +1,6 @@
+1. You should create a variable named `currentDate` and assign it the current date and time using the `Date` object.
+1. You should create a function named `formatDateMMDDYYYY` that takes a parameter. You can name this parameter whatever you like.
+1. The function should use `getMonth`, `getDate`, and `getFullYear` methods to construct the date string. You will need to add 1 to the month method.
+1. You should create a function named `formatDateLong` that takes a parameter. You can name this parameter whatever you like.
+1. Your `formatDateLong` function should return a string that uses the `toLocaleDateString` method to format the date in a `Month Day, Year` format. Remember that the `toLocaleDateString` method accepts `locales` and `options` parameters. (Ex `toLocaleDateString(locales, options)`) For the `locales`, you can pass in `undefined` and for the `options` parameter, you can pass in an object like you learned about in the lecture videos.
+1. You should print the current date and the results of the above functions to the console.
```

Comment:
Wonder if we should be specific about how the logs should be so we can test against them? E.g: `You should print Formatted Date (MM/DD/YYYY):, formatDateMMDDYYYY(currentDate) to the console.` - maybe not worded exactly like that, but something specific anyway. What do you think @larymak?

Reply:
From the previous suggestion, it was highlighted the need to encourage learners to structure their output in a way that makes sense to them. as long as the output correctly reflects the current date and the formatted date strings.

Reply:
I think an easy solution here would be to have campers create a variable called `shortDateFormat` and then have them assign the result to that. Then they just log that variable.

That will make testing way easier and should allow for multiple valid answers like these:

```js
const shortDateFormat = `Formatted Date (MM/DD/YYYY): ${formatDateMMDDYYYY(currentDate)}`;
console.log(formattedDate);
```

```js
const shortDateFormat = 'Formatted Date (MM/DD/YYYY): ' + formatDateMMDDYYYY(currentDate);
```

```js
let someRandomVariableACamperCreated = formatDateMMDDYYYY(currentDate);
let shortDateFormat = 'Formatted Date (MM/DD/YYYY): ' + someRandomVariableACamperCreated;

```

this gives campers options to come up with whatever solution they like as long as it is correct.

same goes longer date format.
have campers create a `longDateFormat` variable, assign the correct result to it, then log it to the console

```js
const longDateFormat = `Formatted Date (Month Day, Year): ${formatDateLong(currentDate)}`;
console.log(longDateFormat);
```

Reply:
as for the user story, it could be something along these lines

```md
Your `shortDateFormat` variable should hold the string `Formatted Date (MM/DD/YYYY): [shortDateFormat goes here]`. You will need to replace `[shortDateFormat goes here]` with the `shortDateFormat`.
```